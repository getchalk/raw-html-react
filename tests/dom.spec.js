/**
 * @jest-environment jsdom
 */

import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import ReactHtml from '../src';
import fixtures from './fixtures';
import FakeElement from './FakeElement';

describe('client side rendering', async () => {
  for (const testName in fixtures) {
    it(testName, () => {
      const html = fixtures[testName];
      let nodeRef;

      const element = <ReactHtml componentMap={{ FakeElement }} html={html} />;

      ReactTestRenderer.create(element, {
        createNodeMock: el => (nodeRef = document.createElement(el.type))
      });

      expect(nodeRef.firstChild).toMatchSnapshot();
    });
  }
});
