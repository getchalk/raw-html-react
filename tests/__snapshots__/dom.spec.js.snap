// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client side rendering Handles comment 1`] = `<!-- comment -->`;

exports[`client side rendering Handles comments inside tags 1`] = `
<ul>
  <!-- &lt;li&gt;test&lt;/li&gt; -->
</ul>
`;

exports[`client side rendering leaves non-react componetns in place 1`] = `
<div>
  <span
    style="color: blue;"
  >
    hello
  </span>
  <p>
    test
  </p>
</div>
`;

exports[`client side rendering parse react component from string 1`] = `
<span
  style="color: red;"
>
  hello
</span>
`;

exports[`client side rendering parse react component from string with props 1`] = `
<span
  style="color: blue;"
>
  hello
</span>
`;

exports[`client side rendering parses html with siblings 1`] = `
<ul
  class="list"
>
  <li>
    Text1
  </li>
  <li>
    Text2
  </li>
</ul>
`;

exports[`client side rendering parses top level siblings without wrapper 1`] = `
<li>
  item-1
</li>
`;

exports[`client side rendering passes through class, width, height, src 1`] = `
<span
  style="color: red;"
>
  hello
</span>
`;

exports[`client side rendering renders text nodes 1`] = `i am pure text`;

exports[`client side rendering supports style 1`] = `
<div
  style="background-color: #fff;"
/>
`;
